### PR TITLE
Big Batch of Cherrypicked Updates

### PR DESCRIPTION
**Pull request by Michiyamenotehifunana**
## About The Pull Request
This PR will optimize the heck out of doors, fix flashbangs, fix and optimize full auto guns, and adds projectile miss chances!  Also allows for the rechambering of handmade guns with a screwdriver.

UPDATE #1: Superior mob optimizations have been ported. Roaches will now react much faster, and spiders are no longer braindead.

## Why It's Good For The Game

Optimizations, bug fixes, balance, and stuff! What's not to love about this?

## Changelog
```changelog Toriate
add: Ported a ton of stuff from upstream that should make full auto guns not break when holstered, and make flashbangs properly disappear when banged. Handmade guns can now also be rechambered with a screwdriver. Superior mob AIs are also much more snappy now.
balance: Bullets can now miss! Chance depends on your VIG stat.
fix: Spiders are no longer braindead.
```

### CODE REVIEW

File: code/game/machinery/adv_med.dm [hunk 1]
```diff
@@ -381,7 +381,9 @@
 					var/obj/item/weapon/implant/device = I
 					other_wounds += "[device.get_scanner_name()] implanted"
 				else
-					unknown_body = TRUE
+					var/obj/item/weapon/implant/device = I
+					if(!device.scanner_hidden)
```
Comment (Jamini) [inline code/game/machinery/adv_med.dm:385]: The implants that have this set to true need those lines uncommented.

File: code/game/machinery/doors/airlock.dm [hunk 2]
```diff
@@ -310,17 +310,7 @@ GLOBAL_LIST_EMPTY(wedge_icon_cache)
 	var/last_event = 0
 
 /obj/machinery/door/airlock/Process()
-	// Deliberate no call to parent.
```
Comment (Jamini) [inline code/game/machinery/doors/airlock.dm:313]: Airlocks are one of the few things that aren't terribly buggy right now. If this breaks airlocks I will be very, very mad. This code is extensively tested by you right Tori?

File: code/game/objects/items/weapons/implant/implants/freedom.dm [hunk 3]
```diff
@@ -51,7 +53,7 @@
 /obj/item/weapon/implant/freedom/on_install(mob/living/source)
 
 	activation_emote = input("Choose activation emote:") in list("blink", "blink_r", "eyebrow", "chuckle", "twitch_s", "frown", "nod", "blush", "giggle", "grin", "groan", "shrug", "smile", "pale", "sniff", "whimper", "wink")
-	uses = rand(2, 5)
+	uses = 1
```
Comment (Jamini) [inline code/game/objects/items/weapons/implant/implants/freedom.dm:56]: This is a massive nerf. Unjustified. We should keep it at 2-5 uses.

File: code/modules/projectiles/guns/projectile/automatic/ak47.dm [hunk 4]
```diff
@@ -204,3 +204,23 @@
 	)
 
 	price_tag = 500
+
+/obj/item/weapon/gun/projectile/automatic/ak47/makeshift/attackby(obj/item/W, mob/user)
```
Comment (Jamini) [inline code/modules/projectiles/guns/projectile/automatic/ak47.dm:208]: Doesn't this make it impossible to remove gunmods from these guns?

**Comment by Shadow-Quill** [comment]
I said this in discord, but I'll put it here as well:
If this pulls over a lot of the file refactors that Eris made to core stuff (ie master controller, ticker, etc), that'd help stop me from needing to pull so many files for https://github.com/Occulus-Server/Occulus-Eris/pull/753.
Even if this pr doesn't add CI, simply getting those files updooted will help. (Basically, don't consider my PR to be blocking this. It's a draft anyways.)

**Comment by Jamini** [comment]
As per our discord discussion, please run a few more test cases on this to make sure nothing breaks.

- Test hacking doors
- Test doors with switches
- Test using a screwdriver on a modified homemade weapon

**Comment by Michiyamenotehifunana** [comment]
Door hacking and external airlocks were completely broken, they've been fixed now.

Screwdrivering an unloaded modified handmade gun will both rechamber and bring up the mod removal screen. Having the gun loaded prevents this.

No issues were found with any other points.

**Comment by Michiyamenotehifunana** [comment]
Also this PR now has the superior mobs optimizations ported, because the door reworks necessitated it as doors now trigger superior mobs. The superior mobs optimizations also required TURF_RANGE stuff to be ported.

As a result, roaches react much faster now, and spiders are no longer lobotomized. This also comes with a claimed 20% reduction in tick usage for the mobs subsystem.

**Review by Jamini — APPROVED** [review]
Thank you  for checking on everything Tori.

**Pull request merged**